networkx/algorithms/shortest_paths/weighted.py

```diff
@@ -1935,3 +1939,200 @@ def dist_path(v):
+    According to the OP, a set of nodes N i is given, each of which is
+    characterized by a score Si. The starting point (node ​​1) and the end point
+    (node ​​N) are data (ie known).The tij transition time from a node i to a node
+    j is given (ie known) for each pair of nodes. All the nodes of the problems
+    cannot be used due to the existence of a time limit Tmax, within which the
+    journey has to be made. The aim of the OP is to determine the route, given
+    the time limit Tmax and the start and end nodes, which will include some of
+    the other nodes of the problem in order to maximize the total collected
+    score. Each node is selected only once, and the overall score will result by
+    summing the scores of the nodes that are selected.
+
+    The algorithm needs an initial solution and at every iteration it selects a
+    neigbor solution.
+
+    Considering c(x) the cost of current solution and c(x') the cost of
+    neighbor solution, then:
+    If c(x') - c(x) <= 0, the neighbor solution becomes current solution for the
+    next iteration. Otherwise, algorithm accepts the neighbor solution to become
+    current solution for the next iteration with probability
+    p = exp - ([c(x') - c(x)] / temp).
+
+    Temp is a parameter of the algorithm that represents temperature in every
+    iteration.
+
+    For more information about Simulated Annealing, see here:
+    https://en.wikipedia.org/wiki/Simulated_annealing
+
+    """
+    if not nx.is_weighted(G):
+        raise nx.NetworkXError('Given graph is not weighted.')
+
+    if not _graph_is_full_completed(G):
+        raise nx.NetworkXError('The graph you have given is not completed.')
+
+    if init_sol is not None:
+        if not _check_feasible_init_sol(G, init_sol, tmax):
+            raise nx.NetworkXError('The initial solution is not feasible.')
+
+    if temp <= 0:
+        raise nx.NetworkXError('The temperature must be greater than 0.')
+
+    best_sol = list(init_sol)
+    current_sol = list(init_sol)
+    for i in range(outer_loop):
+        for j in range(inner_loop):
+            neighbor_sol = list(_moving(G, current_sol, tmax))
+            score_of_current = _score_calculation(scores, current_sol)
+            score_of_neighbor = _score_calculation(scores, neighbor_sol)
+            df = score_of_neighbor - score_of_current
+            if df >= 0:
+                current_sol = list(neighbor_sol)
+                if  _score_calculation(scores, current_sol) > _score_calculation(scores, best_sol):
+                    best_sol = list(current_sol)
+            else:
+                p = float("{0:.2f}".format(exp(df/temp)))
+                r = float("{0:.2f}".format(random()))
+                current_sol = list(neighbor_sol) if r < p else list(current_sol)
+        temp = temp - temp * a
+    best_score = _score_calculation(scores, current_sol)
+    return best_sol, best_score, _cost_calculation(G, best_sol)
+
+def _moving(G, current_sol, tmax):
```

Comment:
This is not a clear name for this function; maybe `_candidate_neighbor` or something similar?